src/main/java/cz/muni/fi/pa165/entity/Hall.java

```diff
@@ -0,0 +1,79 @@
+package cz.muni.fi.pa165.entity;
+
+import java.util.Objects;
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Class for halls' evidence
+ * @author xtrnkal
+ */
+@Entity
+public class Hall {
+  
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    @NotNull
+    private String name;
+
+    @Column(nullable = false)
+    @NotNull
+    private Long capacity;
+
+    @Column(nullable = false)
+    private String address;
+
+    public Hall(String name, Long capacity) {
+        this.name = name;
+        this.capacity = capacity;
+    }
+    
+    public Long getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Long getCapacity() {
+        return capacity;
+    }
+
+    public void setCapacity(Long capacity) {
+        this.capacity = capacity;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, capacity);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof Hall)) {
+            return false;
+        }
+        Hall other = (Hall) obj;
+        return Objects.equals(name, other.getName()) && Objects.equals(capacity, other.getCapacity());
```

Comment:
Pokud je porovnavana jedna polozka, tak fajn, ale kdyz je jich vic, tak je lepsi dat if(!.....) return false pro kazdou polozku a pak return true na konci. Neni to vubec nic o cistote kodu nebo jak to nazvat, ale zkus si nad tim zapnout debugger a zacnes to tak brzo psat.

Reply:
Chápu, díky :)